ballerina/googlesheets_client.bal

```diff
@@ -545,4 +555,54 @@ public isolated client class GoogleSheetsClient {
         }
     }
 
+    private isolated function sendRequest(http:Client httpClient, string path) returns json | error {
+        http:Response|error httpResponse = httpClient->get(path);
+        if httpResponse is http:Response {
+            int statusCode = httpResponse.statusCode;
+            json | error jsonResponse = httpResponse.getJsonPayload();
+            if jsonResponse is json {
+                error? validateStatusCodeRes = self.validateStatusCode(jsonResponse, statusCode);
+                if (validateStatusCodeRes is error) {
+                    return validateStatusCodeRes;
+                }
+                return jsonResponse;
+            } else {
+                return self.getSpreadsheetError(jsonResponse);
+            }
+        } else {
+            return self.getSpreadsheetError(<json|error>httpResponse);
+        }
+    }
+
+    private isolated function validateStatusCode(json response, int statusCode) returns error? {
+        if statusCode != http:STATUS_OK {
+            return self.getSpreadsheetError(response);
+        }
+    }   
+    isolated function getSpreadsheetError(json|error errorResponse) returns error {
+        if errorResponse is json {
+                return error(errorResponse.toString());
+        } else {
+                return errorResponse;
+        }
+    }
```

Comment:
code formatting issue.

Reply:
we don't need else block here. we can just return `errorResponse`

Reply:
Updated in [afd480a](https://github.com/ballerina-platform/module-ballerinax-persist.googlesheets/pull/40/commits/afd480aa5bbc7bf276898d8223c0e1885bd8d34c)